### ISSUE TITLE
Running local on MacOsX El Capitain

### ISSUE THREAD

**Issue by nxbdi** [SnappyDataInc/snappydata#183]
snappy> connect client 'localhost:1527';
ERROR 08006: (SQLState=08006 Severity=40000) A communications error has been detected: Failed after trying all available servers: [], for control URL: localhost[1527].
Caused by: ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
ERROR 08006: (SQLState=08006 Severity=40000) A communications error has been detected: Failed after trying all available servers: [localhost[1527]], for control host[port]: localhost[1527].
Caused by: ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
ERROR 08001: (SQLState=08001 Severity=40000) java.net.ConnectException : Error connecting to server localhost on port 1527 with message Connection refused.
Caused by: ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)

16/02/18 14:17:39.384 EST serverConnector<tid=0xc> INFO snappystore: Initializing region _DDL_STMTS_META_REGION
16/02/18 14:17:39.385 EST serverConnector<tid=0xc> INFO snappystore: Region /_DDL_STMTS_META_REGION recovered from the local disk. Old persistent ID: /127.0.0.1:/Users/timothyspann/Downloads/snappydata-0.1.0-PREVIEW/work/localhost-locator-1/./datadictionary created at timestamp 1455822888412 version 0 diskStoreId bd4a1e44-f164-46d2-8855-281e40271e87 name , new persistent ID /127.0.0.1:/Users/timothyspann/Downloads/snappydata-0.1.0-PREVIEW/work/localhost-locator-1/./datadictionary created at timestamp 1455823059383 version 0 diskStoreId bd4a1e44-f164-46d2-8855-281e40271e87 name
16/02/18 14:17:40.018 EST serverConnector<tid=0xc> INFO snappystore: FabricDatabase: waiting for index loading from GFXD-DEFAULT-DISKSTORE
16/02/18 14:17:40.019 EST serverConnector<tid=0xc> INFO snappystore: FabricDatabase: Index loading completed for GFXD-DEFAULT-DISKSTORE in 0 ms
16/02/18 14:17:40.021 EST Idle OplogCompactor<tid=0x37> INFO snappystore: Oplog::recoverValuesIfNeeded: recovering values from oplog#1
16/02/18 14:17:40.317 EST serverConnector<tid=0xc> INFO snappystore: FabricDatabase: initial DDL replay completed.
16/02/18 14:17:40.317 EST serverConnector<tid=0xc> INFO snappystore: FabricDatabase: Authentication recheck successful.
16/02/18 14:17:40.465 EST main<tid=0x1> INFO snappystore: TraceFabricServiceBoot: Starting DRDA server on: localhost/127.0.0.1[1527]
SnappyData Network Server - 10.4.2.0 beta - (1) started and ready to accept connections on port 1527 at 2016/02/18 14:17:40.500 EST
16/02/18 14:17:40.501 EST gemfirexd.NetworkServerStarter<tid=0x3d> INFO snappystore: SnappyData Network Server - 10.4.2.0 beta - (1) started and ready to accept connections on port 1527 at 2016/02/18 14:17:40.500 EST
16/02/18 14:17:40.509 EST NetworkServerThread_13<tid=0x3e> INFO snappystore: (tid=62 msgId=0) Failed to set OPT_KEEPINTVL on socket: setSocketOption(): socket option OPT_KEEPINTVL not supported by current platform MACOSX
16/02/18 14:17:40.509 EST NetworkServerThread_13<tid=0x3e> INFO snappystore: (tid=62 msgId=1) Failed to set OPT_KEEPCNT on socket: setSocketOption(): socket option OPT_KEEPCNT not supported by current platform MACOSX
16/02/18 14:17:40.540 EST main<tid=0x1> INFO snappystore: TraceFabricServiceBoot: Started DRDA server on: localhost/127.0.0.1[1527]

➜  snappydata-0.1.0-PREVIEW  ./sbin/snappy-start-all.sh
localhost: Starting SnappyData Locator using peer discovery on: localhost[10334]
localhost: Starting DRDA server for SnappyData at address localhost/127.0.0.1[1527]
localhost: Logs generated in /Users/timothyspann/Downloads/snappydata-0.1.0-PREVIEW/sbin/../work/localhost-locator-1/gfxdlocator.log
localhost: SnappyData Locator pid: 78508 status: running
localhost: zsh:1: no matches found: -locators=timspannspark.local[10334]
localhost: zsh:1: no matches found: -locators=timspannspark.local[10334]

**Comment by ymahajan** [comment]
Are server and locator started with default configuration ? Could you please share server/locator configuration files? There should a stack-trace in the server logs. Is 'timspannspark.local' a known host? Is this an intermittent or consistent failure ?

**Comment by radutanasa** [comment]
> ➜ snappydata-0.1.0-PREVIEW ./sbin/snappy-start-all.sh
> localhost: Starting SnappyData Locator using peer discovery on: localhost[10334]
> localhost: Starting DRDA server for SnappyData at address localhost/127.0.0.1[1527]
> localhost: Logs generated in /Users/timothyspann/Downloads/snappydata-0.1.0-PREVIEW/sbin/../work/localhost-locator-1/gfxdlocator.log
> localhost: SnappyData Locator pid: 78508 status: running
> localhost: zsh:1: no matches found: -locators=timspannspark.local[10334]
> localhost: zsh:1: no matches found: -locators=timspannspark.local[10334]

I got this as well while using zsh. Switching to bash fixed it.

**Comment by sumwale** [comment]
@radutanasa Thanks. Perhaps there is some implicit dependence on bash in the scripts.

@hbhanawat can you check the scripts to see: either a) all explicit depend on bash, or b) don't have bashisms. Need to determine the cause of original error which is tough without logs, but the above information can be useful (i.e. try to reproduce in zsh). Can we track this to be fixed for next release?

**Comment by radutanasa** [comment]
@sumwale I'll try to looking into it as well.

**Comment by radutanasa** [comment]
I managed to reproduce this issue. 

1. Run `chsh -s /bin/zsh`
2. Restart the Terminal
3. Run `./sbin/snappy-start-all`

The resulting error is:

```
localhost: Starting SnappyData Locator using peer discovery on: localhost[10334]
localhost: Starting DRDA server for SnappyData at address localhost/127.0.0.1[1527]
localhost: Logs generated in /Users/radu/Development/programs/snappydata-0.1.0-PREVIEW/sbin/../work/localhost-locator-1/gfxdlocator.log
localhost: SnappyData Locator pid: 10109 status: running
localhost: zsh:1: no matches found: -locators=RaduAdlgicasMBP[10334]
localhost: zsh:1: no matches found: -locators=RaduAdlgicasMBP[10334]
```

To go around this issue, run `chsh -s /bin/bash` and restart the Terminal.

**Comment by radutanasa** [comment]
@hbhanawat @sumwale OK, I found the problem. I'll make a pull request.

**Comment by sumwale** [comment]
Fixed in ba74b33 courtesy @radutanasa
